bbl_screen-patch/kexec_ui/printerui/qml/InstallingPage.qml

```diff
@@ -92,13 +92,19 @@ Rectangle {
                 X1PlusNative.system('while true ; do iwconfig wlan0 power off > /dev/null 2>&1 ; sleep 4 ; done &');
             }
 
-            var rv = X1PlusNative.system(`mkdir -p /userdata/x1plus && cd /userdata/x1plus && unzip -p /sdcard/${x1pName} payload.tar.gz | gunzip | tar xv`);
-            // We should probably error check this somehow -- do an XHR to make sure the launch script is there?
-            statusModel[statusModel.length - 1][0] = "success";
-            statusModel = statusModel;
-            secondaryStatusText = "";
-            X1PlusNative.system('mount -o remount,exec /userdata'); // Firmware R mounts it noeexec
-            X1PlusNative.system(`/userdata/x1plus/install.sh &`); // this will soon start communicating with us over DDS, hopefully
+            var rv = X1PlusNative.popen(`mkdir -p /userdata/x1plus && cd /userdata/x1plus && unzip -p /sdcard/${x1pName} payload.tar.gz | gunzip | tar xv`);
+            let install_path = X1PlusNative.getenv("EMULATION_WORKAROUNDS") + "/userdata/x1plus/install.sh"
+            let resp = X1PlusNative.readFile(install_path);
+            if (resp.byteLength !== 0 && rv !== 0) { /* verify that install.sh exists and.. isn't blank? */
+                statusModel[statusModel.length - 1][0] = "success";
+                statusModel = statusModel;
+                secondaryStatusText = "";
+                X1PlusNative.system('mount -o remount,exec /userdata'); // Firmware R mounts it noeexec
+                X1PlusNative.system(`/userdata/x1plus/install.sh &`); // this will soon start communicating with us over DDS, hopefully
+            } else {
+                console.log("[x1p] failed to unpack x1p file");
+                dialogStack.replace("SelectX1pPage.qml"); //send user back to selection page
+            }
```

Comment:
I think without a visible diagnostic, it is not obvious to me that just bouncing backwards is any more 'obvious' as to what is going on?  maybe the answer is just to push a message saying "X1P extraction failed.  The X1P file may be corrupt.  Replace the X1P file on the SD card and try again.'?

Reply:
There are several spots in the installer where it would be very nice to have a dialog prompt with a short description of the error and the dialog options are "Back" to the selection screen and "Reboot"